tests/core/data/test_aws_s3_data_node.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+from dataclasses import dataclass
+from unittest.mock import patch
+
+import boto3
+from moto import mock_s3
+import pytest
+
+from src.taipy.core.data.data_node_id import DataNodeId
+from src.taipy.core.data.aws_s3 import S3ObjectDataNode
+from src.taipy.core.exceptions.exceptions import InvalidCustomDocument, MissingRequiredProperty
+from taipy.config.common.scope import Scope
+
+
+
+class TestS3ObjectDataNode:
+    __properties = [
+        {
+            "aws_access_key": "testing",
+            "aws_secret_acces_key": "testing",
+            "aws_region": "us-east-1",
+            "aws_s3_bucket_name":"taipy",
+            "aws_s3_object_key":"taipy-object",
+            "db_extra_args": {},
+        }
+    ]
+
+    @mock_s3
+    @pytest.mark.parametrize("properties", __properties)
+    def test_create(self, properties):
+        aws_s3_object_dn = S3ObjectDataNode(
+            "foo_bar_aws_s3",
+            Scope.SCENARIO,
+            properties=properties,
+        )
+        assert isinstance(aws_s3_object_dn, S3ObjectDataNode)
+        assert aws_s3_object_dn.storage_type() == "aws_s3"
+        assert aws_s3_object_dn.config_id == "foo_bar_aws_s3"
+        assert aws_s3_object_dn.scope == Scope.SCENARIO
+        assert aws_s3_object_dn.id is not None
+        assert aws_s3_object_dn.owner_id is None
+        assert aws_s3_object_dn.job_ids == []
+        assert aws_s3_object_dn.is_ready_for_reading
+
+
+    @mock_s3
+    @pytest.mark.parametrize('data', [('Hello, world!'),])
+    @pytest.mark.parametrize("properties", __properties)
+    def test_write(self, properties, data ):
```

Comment:
This test is no different from the `test_read` method.

The test should do the following:
1. Create S3ObjectDataNode and write data to the data node.
2. Connect to the bucket and read data using boto3.
3. Check if data stored in the bucket is the exact data that written to the S3ObjectDataNode.

If not, the test should just be renamed as `test_read_write`.